File: 118.pascals-triangle.rb
#
# @lc app=leetcode id=118 lang=ruby
#
# [118] Pascal's Triangle
#

# @lc code=start
# @param {Integer} num_rows
# @return {Integer[][]}
def generate(num_rows)
  list = [[1]]

  (1...num_rows).each do |i|
    row = [1]
    row[i] = 1
    prev_row = list[i - 1]

    (1...prev_row.length).each do |j|
      row[j] = prev_row[j - 1] + prev_row[j]
    end
    list[i] = row
  end

  list
end
# @lc code=end

File: 118.pascals-triangle.ts
/*
 * @lc app=leetcode id=118 lang=typescript
 *
 * [118] Pascal's Triangle
 */

// @lc code=start
function generate(numRows: number): number[][] {
  const triangle = [[1]];

  for (let i = 1; i < numRows; i++) {
    const prevRow = triangle[i - 1];
    const row = [1];
    let j = 1;

    for (; j < prevRow.length; j++) {
      row[j] = prevRow[j - 1] + prevRow[j];
    }
    row[j] = 1;
    triangle[i] = row;
  }

  return triangle;
}
// @lc code=end
